src/gui/updateurldialog.h

```diff
@@ -6,27 +6,19 @@
 
 namespace OCC {
 
-class UpdateUrlDialog : public QObject
+class UpdateUrlDialog : public QMessageBox
 {
     Q_OBJECT
 public:
-    explicit UpdateUrlDialog(const QString &messageBoxTitle, const QString &messageBoxContent, const QUrl &oldUrl, const QUrl &newUrl, QWidget *parent = nullptr);
+    explicit UpdateUrlDialog(const QString &title, const QString &content, const QUrl &oldUrl, const QUrl &newUrl, QWidget *parent = nullptr);
 
+    // warning: this method overwrites the superclass's start method
+    // do *not* call the function via a base class pointer!
     void show();
```

Comment:
Thats a hack, I assume we can move all logic to the constructor and use the base implementation of show?

Reply:
How would you signalize that "immediate return" from the constructor? You need to be able to hook up a method to `accepted` before the URLs are compared. We can't use exceptions there.

I was thinking about a named constructor with a `bool` return type.

Reply:
I'll try that.